src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -399,53 +399,105 @@ fn make_core(
   )
 }
 
-fn workunits_to_py_tuple_value<'a>(workunits: impl Iterator<Item = &'a WorkUnit>) -> Value {
-  let workunit_values = workunits
-    .map(|workunit: &WorkUnit| {
-      let mut workunit_zipkin_trace_info = vec![
-        (
-          externs::store_utf8("name"),
-          externs::store_utf8(&workunit.name),
-        ),
-        (
-          externs::store_utf8("start_secs"),
-          externs::store_u64(workunit.time_span.start.secs),
-        ),
-        (
-          externs::store_utf8("start_nanos"),
-          externs::store_u64(u64::from(workunit.time_span.start.nanos)),
-        ),
-        (
-          externs::store_utf8("duration_secs"),
-          externs::store_u64(workunit.time_span.duration.secs),
-        ),
-        (
-          externs::store_utf8("duration_nanos"),
-          externs::store_u64(u64::from(workunit.time_span.duration.nanos)),
-        ),
-        (
-          externs::store_utf8("span_id"),
-          externs::store_utf8(&workunit.span_id),
-        ),
-      ];
-      if let Some(parent_id) = &workunit.parent_id {
-        workunit_zipkin_trace_info.push((
-          externs::store_utf8("parent_id"),
-          externs::store_utf8(parent_id),
-        ));
-      }
+fn started_workunit_to_py_value(started_workunit: &StartedWorkUnit) -> Value {
+  use std::time::UNIX_EPOCH;
+  let duration = started_workunit
+    .start_time
+    .duration_since(UNIX_EPOCH)
+    .unwrap_or_else(|_| Duration::default());
```

Comment:
Why use the default duration? Should we blow up in this case because there's an error in the workunits code, rather than silently using a default that will behave unexpectedly?